File: javascript/str_join.js
const chunk = "regexp123abcbcd";
const repeatCount = 10000;
const size = 1000;

function join() {
    let str = "";
    for (let j = 0; j < size; ++j) str += chunk;
}

const start = Date.now();
for (let i = 0; i < repeatCount; ++i) {
    join();
}
const end = Date.now();

console.log(end - start);

File: c++/str_join.cpp
#include <iostream>
#include <chrono>
#include <cstring>
using namespace std::chrono;

const char chunk[] = "regexp123abcbcd";
const int repeat_count = 10000;
const int size = 1000;

void join() {
    char *str = new char();
    str[0] = '\0';
    for (auto i = 0; i < size; ++i) strcat(str, chunk);
}

int main() {
    auto start = system_clock::now();
    for (auto i = 0; i < repeat_count; ++i) {
        join();
    }
    auto end = system_clock::now();
    std::cout << duration_cast<milliseconds>(end - start).count() << std::endl;
    return 0;
}
